node_modules/oae-content/lib/api.js

```diff
@@ -997,8 +1019,7 @@ var setContentPermissions = module.exports.setContentPermissions = function(ctx,
     validator.check(contentId, {'code': 400, 'msg': 'A content id must be provided'}).isResourceId();
     validator.check(_.keys(permissionChanges).length, {'code': 400, 'msg': 'You should specify at least 1 user/group to set content permissions on'}).min(1);
     _.each(permissionChanges, function(role, principalId) {
-        validator.check(principalId, {'code': 400, 'msg': 'A principalId needs to be specified for a role.'}).isPrincipalId();
```

Comment:
Should we create a validator function that checks whether a string is either a principalId or an email address?